### PR TITLE
New CI Runners

### PR DESCRIPTION
**Pull request by muellerzr**
Migrates CI runners to use AWS

Some notes:

* Use `[self-hosted, intel-cpu, 8-cpu, ci]` to utilize a CPU runner. These aren't stored in our local runners and are spun up by demand, hence why they do not show up in settings
* Use `[self-hosted, docker-gpu]` for GPU runners. (We can also fine-tune these using `push-ci` and `nightly-ci` for whether it's a nightly run or pushing to main)
* In general now these CI runners assume no persistent storage to make things easier

### CODE REVIEW

File: .github/workflows/build-docker-images-release.yml [hunk 1]
```diff
@@ -41,7 +41,7 @@ jobs:
 
   version-cuda:
     name: "Latest Accelerate GPU [version]"
-    runs-on: [self-hosted, docker-gpu, multi-gpu, gcp]
+    runs-on: [self-hosted, intel-cpu, 8-cpu, ci]
```
Comment (ydshieh) [inline .github/workflows/build-docker-images-release.yml:44]: a bit strange that here we use cpu machine while the job name is GPU

File: .github/workflows/self_hosted_integration_tests.yml [hunk 2]
```diff
@@ -74,6 +74,10 @@ jobs:
         run: |
           source activate accelerate;
           pytest -sv tests/deepspeed
+      - name: Clean local files
+        if: always()
+        run: |
+          rm -rf accelerate transformers
 
       - name: Run transformers examples tests
         working-directory: transformers/
```
Comment (ydshieh) [inline .github/workflows/self_hosted_integration_tests.yml:83]: quick look: this seems a bit odd, where we delete `transformers` but having `working-directory: transformers` ..?

**Review by ydshieh — APPROVED** [review]
overall lgtm, but having 2 questions that confuse me ..?

File: .github/workflows/self_hosted_integration_tests.yml (hunk 2 above)
Comment (ydshieh) [reply .github/workflows/self_hosted_integration_tests.yml:83]: my 2 comments may apply to multiple places (if they make sense)

File: .github/workflows/build-docker-images-release.yml (hunk 1 above)
Comment (muellerzr) [reply .github/workflows/build-docker-images-release.yml]: Fixed!

File: .github/workflows/self_hosted_integration_tests.yml (hunk 2 above)
Comment (muellerzr) [reply .github/workflows/self_hosted_integration_tests.yml]: Yes we now have non-persistent workers so this shouldn't be necessary.

File: .github/workflows/build-docker-images-release.yml (hunk 1 above)
Comment (glegendre01) [reply .github/workflows/build-docker-images-release.yml]: i agree, but it was docker-cpu...

Comment (glegendre01) [reply .github/workflows/build-docker-images-release.yml]: it's for building image. so maybe no need GPU ?

File: .github/workflows/build-docker-images-release.yml [hunk 3]
```diff
@@ -41,7 +41,7 @@ jobs:
 
   version-cuda:
     name: "Latest Accelerate GPU [version]"
-    runs-on: [self-hosted, docker-gpu, multi-gpu, gcp]
+    runs-on: [self-hosted, docker-gpu]
```
Comment (glegendre01) [inline .github/workflows/build-docker-images-release.yml:44]: do you need GPU here ? if yes also need to use [self-hosted, single-gpu, nvidia-gpu, t4, daily-ci]

File: .github/workflows/build_docker_images.yml [hunk 4]
```diff
@@ -41,8 +31,7 @@ jobs:
 
   latest-cuda:
     name: "Latest Accelerate GPU [dev]"
-    runs-on: [self-hosted, docker-gpu, multi-gpu, gcp]
-    needs: clean-storage
+    runs-on: [self-hosted, docker-gpu]
     steps:
```
Comment (glegendre01) [inline .github/workflows/build_docker_images.yml:35]: same here

**Pull request merged**